sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -376,7 +383,14 @@ public virtual Response<DetectLanguageResultCollection> DetectLanguageBatch(IEnu
                 {
                     case 200:
                         IDictionary<string, int> map = CreateIdToIndexMap(documents);
-                        return CreateDetectLanguageResponse(response, map);
+                        DetectLanguageResultCollection results = CreateDetectLanguageResponse(response, map);
+                        if (results[0].HasError && string.IsNullOrEmpty(results[0].Id))
+                        {
+                            // InvalidDocumentBatch.
```

Comment:
it also might be better to pull this out into a function that they each call to avoid duplicating code